### PR TITLE
Add support for formatting DateTimeImmutable

### PR DESCRIPTION
**Pull request by thetrompf**

### CODE REVIEW

File: src/Monolog/Formatter/LogglyFormatter.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ public function __construct($batchMode = self::BATCH_MODE_NEWLINES, $appendNewli
      */
     public function format(array $record)
     {
-        if (isset($record["datetime"]) && ($record["datetime"] instanceof \DateTime)) {
+        if (isset($record["datetime"]) && ($record["datetime"] instanceof \DateTimeInterface)) {
```
Comment (xabbuh) [inline src/Monolog/Formatter/LogglyFormatter.php]: This won't work with PHP 5.3 and 5.4. The `DateTimeInterface` was introduced in PHP 5.5.

Comment (thetrompf) [inline src/Monolog/Formatter/LogglyFormatter.php]: Oh my bad, I hadn't checked up on that

**Comment by Seldaek** [comment]
Indeed, let's keep this around for v2 which I believe will require 5.5 or above

**Comment by stof** [comment]
@Seldaek just add instanceof checks for both DateTime and DateTimeInterface and it will keep working on 5.3 and 5.4 while supporting 5.5+ fully.

**Comment by Seldaek** [comment]
Yeah that's alright for instanceof but my concern was more about the type hints. We can just remove the hints but arguably that'd make it a BC break...

**Comment by ZergMark** [comment]
This should be fine now that 2.x is PHP7+

**Comment by Seldaek** [comment]
Replaced by https://github.com/Seldaek/monolog/pull/793

**Pull request closed without merging**